compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1448,8 +1448,16 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_expr_path_start(&mut self) -> PResult<'a, P<Expr>> {
+        let maybe_eq_tok = self.prev_token.clone();
         let (qself, path) = if self.eat_lt() {
-            let (qself, path) = self.parse_qpath(PathStyle::Expr)?;
+            let lt_span = self.prev_token.span;
+            let (qself, path) = self.parse_qpath(PathStyle::Expr).map_err(|mut err| {
```

Comment:
This could use a comment explaining what this logic is for. I don't think it necessarily needs to be pulled out into its own like `Parser::suggest_*` fn, but it could be useful to motivate why we're doing this with some text.